### ISSUE TITLE
Does it allow unsubscribing directly?

### ISSUE THREAD

**Issue by deanrad** [zzarcon/react-observable-hook#1]
I made this CodePen to show how I'm using `react-observable-hook` to provide an interface to Observables similar to the way Apollo client for GraphQL - allows Query Components to give `{data, loading, error}` props to UI components. 

https://codesandbox.io/s/v6ln99zmx0

I don't mind interrupting the original Observable to make it cancel, but I think effect hooks can support cancellation functions.. Just wanted to know your thoughts.

**Comment by zzarcon** [comment]
I see, if I get it right you would like to have the `useObservable` to return a **cancel**, that you can just hook in your button for instance?

```
const {result, cancel} = useObservable(observable, () => ({
  loading: false
}));
  
<button onClick={cancel}>Cancel</button>
```

Let me know if this is what you are after :)

**Comment by deanrad** [comment]
Yes I was pondering what return value style would support that, but that's exactly the idea :)
